libs/geometry/src/TangentSpaceMesh.cpp

```diff
@@ -186,15 +172,130 @@ Algorithm selectAlgorithm(TangentSpaceMeshInput *input) {
 
     return outAlgo;
 }
-
 #undef IS_INPUT_TYPE
 
+// The paper uses a Z-up world basis, which has been converted to Y-up here
+void frisvadKernel(const float3& N, float3& T, float3& B) {
```

Comment:
should all these function be `noexcept`?

Reply:
done